cloudinit/distros/freebsd.py

```diff
@@ -612,18 +222,12 @@ def apply_locale(self, locale, out_fn=None):
                 util.logexc(LOG, "Failed to restore %s backup",
                             self.login_conf_fn)
 
-    def _bring_up_interface(self, device_name):
-        if device_name.startswith('lo'):
-            return
-        dev = self.getnetifname(device_name)
-        cmd = ['/etc/rc.d/netif', 'start', dev]
-        LOG.debug("Attempting to bring up interface %s using command %s",
-                  dev, cmd)
-        # This could return 1 when the interface has already been put UP by the
-        # OS. This is just fine.
-        (_out, err) = util.subp(cmd, rcs=[0, 1])
-        if len(err):
-            LOG.warning("Error running %s: %s", cmd, err)
+    def apply_network_config_names(self, netconfig):
+        # This is handled by the freebsd network renderer. It writes in
+        # /etc/rc.conf a line with the following format:
+        #    ifconfig_OLDNAME_name=NEWNAME
+        # FreeBSD network script will rename the interface automatically.
+        return
```

Comment:
I wonder if a log message might be handy here. If, for example, there's some sort of race condition to do with interface renaming (which presumably happens later on FreeBSD than Linux), it could be very helpful.

Reply:
I added a log here:

https://github.com/canonical/cloud-init/pull/61/commits/645c279571655a0a1b35a6ae10c4c5bccd1a3955#diff-d95efef08d84ea4931c64c089b56087eR38-R39